### DISCUSSION TITLE
"set-env" depracation warning when running "dotnet build"

### DISCUSSION THREAD

**Discussion by Eran-BenShahar** [community/community#25011] [Actions]
<p>My workflow keeps getting these deprecation warning messages.<br>
I upgraded all used actions to newest versions (actions/checkout@v2.2.3 and actions/setup-dotnet@v1.7.2) and most of the warnings disappeared.<br>
Now I only get these warnings on a simple “run” step which looks like this:</p>
<ul>
<li>name: Build<br>
run: dotnet build src/Main/Main.csproj --configuration Release<br>
working-directory: ./</li>
</ul>
<p>How can I figure out what is the issue?<br>
I’m simply calling “dotnet build”.</p>
<p>Also, I do declare 2 environment variables using the “env:” syntax at the workflow level. Could that be the issue?</p>

**Comment by Simran-B** [comment]
<p><code>env:</code> is completely unrelated. Only <code>::set-env</code> is deprecated.</p>
<p>Many actions depend on <code>@actions/core</code>. It implemented the new way to set env vars via environment files in v1.2.6. Several official actions still use an older version however, and may thus cause the deprecation warning to show.</p>
<p>See e.g. <a href="https://github.com/actions/checkout/issues/380">https://github.com/actions/checkout/issues/380</a></p>

**Comment by Eran-BenShahar** [comment, marked as answer]
<p>I finally found the issue (sorry for not circling back sooner), it was a NuGet we use during build (<a href="https://Nerdbank.GitVersioning" rel="noopener nofollow ugc">https://github.com/dotnet/Nerdbank.GitVersioning</a>).<br>
It was using <code>set-env</code> internally.<br>
The solution was simply to update it (since they solved it in a later version).</p>
